sonic_data_client/non_db_client.go

```diff
@@ -246,6 +279,39 @@ func getProcStat() ([]byte, error) {
 	return b, nil
 }
 
+func getBuildVersion() ([]byte, error) {
+
+	// Load and parse the content of version file
+	versionFileStash.once.Do(func() {
+		versionFileStash.versionInfo.BuildVersion = "sonic.NA"
+
+		fileContent, err := ImplIoutilReadFile(SonicVersionFilePath)
+		if err != nil {
+			log.Errorf("Failed to read '%v', %v", SonicVersionFilePath, err)
```

Comment:
Wondering will client see actual error if file does not exist for any reason?

Reply:
Added new member `error` to transfer error message in case file reading fails. Empty error message means file read was successful.